front/components/vaults/VaultResourcesList.tsx

```diff
@@ -171,13 +171,49 @@ export const VaultResourcesList = ({
     );
   }
 
-  const setUpDataSources = vaultDataSourceOrViews.map((dsv) => dsv.connectorId);
-  const unusedDataSources = managedDataSources.filter(
-    (ds) => !setUpDataSources.includes(ds.connectorId)
-  );
+  const saveSelectedDatasources = async (
+    selectedDataSources: VaultSelectedDataSources
+  ) => {
+    const body = {
+      content: selectedDataSources.map((sDs) => ({
+        dataSource: sDs.name,
+        parentsIn: sDs.parentsIn,
+      })),
+    };
+
+    const res = await fetch(`/api/w/${owner.sId}/vaults/${vault.sId}`, {
```

Comment:
This should call the `/api/w/${owner.sId}/vaults/${vault.sId}/data_source_views/${dsvId}` (endpoint will be available very soon).